pkg/controller/controller.go

```diff
@@ -383,28 +383,34 @@ func AttachmentPods(ownerPod *k8sv1.Pod, podInformer cache.SharedIndexInformer)
 }
 
 func ApplyNetworkInterfaceRequestOnVMISpec(vmiSpec *v1.VirtualMachineInstanceSpec, request *v1.VirtualMachineInterfaceRequest) *v1.VirtualMachineInstanceSpec {
-	existingIface := vmispec.FilterInterfacesSpec(vmiSpec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
-		return iface.Name == request.AddInterfaceOptions.Name
-	})
-
-	if len(existingIface) == 0 {
-		newNetwork := v1.Network{
-			Name: request.AddInterfaceOptions.Name,
-			NetworkSource: v1.NetworkSource{
-				Multus: &v1.MultusNetwork{
-					NetworkName: request.AddInterfaceOptions.NetworkAttachmentDefinitionName,
-				},
-			},
+	switch {
+	case request.AddInterfaceOptions != nil:
+		if iface := vmispec.LookupInterfaceByName(vmiSpec.Domain.Devices.Interfaces, request.AddInterfaceOptions.Name); iface != nil {
```

Comment:
shouldn't it be `iface == nil`?

Reply:
I think it got fixed in the wrong commit.